### PR TITLE
Support samza portable UDF metrics.

### PR DESCRIPTION
**Pull request by ryucc**
Major fixes: 
In #25068, we tried to support UDF metrics with the samza portable runner, but we were missing a step to copy beam metrics into samza metrics.

This change has been tested on the linkedin/beam mirror, and confirmed the metrics are working now.

Other fixes:
Cleaning up logs: lower a log line from `warn` to `debug`.
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java [hunk 1]
```diff
@@ -148,7 +151,10 @@ private void parseAndUpdateMetric(MetricsApi.MonitoringInfo monitoringInfo) {
         break;
 
       default:
-        LOG.warn("Unsupported metric type {}", monitoringInfo.getType());
+        if (LOG.isDebugEnabled()) {
```
Comment (mynameborat) [inline runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java:154]: IMO, this check is overkill as the log message itself is not constructing complex statements (string)? I think it is better to just do `LOG.debug`  

Additionally it is just more readable than this additional branching.

Comment (kw2542) [reply runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java:154]: +1, LOG.debug() checks debug enabled itself.

File: runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java [hunk 2]
```diff
@@ -90,7 +90,9 @@ public void onProgress(BeamFnApi.ProcessBundleProgressResponse progress) {}
   public void onCompleted(BeamFnApi.ProcessBundleResponse response) {
     response.getMonitoringInfosList().stream()
         .filter(monitoringInfo -> !monitoringInfo.getPayload().isEmpty())
-        .forEach(this::parseAndUpdateMetric);
+        .map(this::parseAndUpdateMetric)
+        .distinct()
+        .forEach(samzaMetricsContainer::updateMetrics);
```
Comment (kw2542) [inline runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java:95]: why do we need to update the metrics again here? IIRC, beam native counter, gauge and distribution metric is integrated with samza metrics already

Comment (ryucc) [reply runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java:95]: Metrics doesn't work without this line in local testing.

samzaMetricsContainer probably has copies of both a samza metrics type and a beam metrics type. This line triggers the copy from beam to samza metrics

File: runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java (hunk 1 above)
Comment (ryucc) [reply runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java:154]: Yes it's overkill.

Comment (ryucc) [reply runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java]: fixed

**Review by xinyuiscool — APPROVED** [review]
LGTM!

**Pull request merged**